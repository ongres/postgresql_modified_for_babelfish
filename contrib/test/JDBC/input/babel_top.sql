--
-- Tests for TOP clause
--

create table students (fname varchar(10), lname varchar(10), score double precision)
go

insert into students (fname, lname, score)
values
    ('John', 'Doe', 72.5),
    ('Jane', 'Smith', 88),
    ('Jill', 'Johnson', 98),
    ('Jack', 'Green', 67),
    ('Jennifer', 'Ross', 75.7),
    ('Jacob', 'Brown', 95.2)
go

select top 3 * from students
go

select top 4 score from students
go

select top 3 from students
go

-- test top bigint
select top 2147483648 * from students
go

-- test top 100 percent
select top 100 percent * from students
go

select top 100.00 percent lname from students
go

-- cleanup
drop table students
go