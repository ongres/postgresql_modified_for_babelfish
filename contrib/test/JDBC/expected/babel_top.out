
--
-- Tests for TOP clause
--
create table students (fname varchar(10), lname varchar(10), score double precision)
go

insert into students (fname, lname, score)
values
    ('John', 'Doe', 72.5),
    ('Jane', 'Smith', 88),
    ('Jill', 'Johnson', 98),
    ('Jack', 'Green', 67),
    ('Jennifer', 'Ross', 75.7),
    ('Jacob', 'Brown', 95.2)
go
~~ROW COUNT: 6~~


select top 3 * from students
go
~~START~~
varchar#!#varchar#!#float
John#!#Doe#!#72.5
Jane#!#Smith#!#88.0
Jill#!#Johnson#!#98.0
~~END~~


select top 4 score from students
go
~~START~~
float
72.5
88.0
98.0
67.0
~~END~~


select top 3 from students
go
~~ERROR (Code: 33557097)~~

~~ERROR (Message: syntax error near 'from' at line 1 and character position 13)~~


-- test top bigint
select top 2147483648 * from students
go
~~START~~
varchar#!#varchar#!#float
John#!#Doe#!#72.5
Jane#!#Smith#!#88.0
Jill#!#Johnson#!#98.0
Jack#!#Green#!#67.0
Jennifer#!#Ross#!#75.7
Jacob#!#Brown#!#95.2
~~END~~


-- test top 100 percent
select top 100 percent * from students
go
~~START~~
varchar#!#varchar#!#float
John#!#Doe#!#72.5
Jane#!#Smith#!#88.0
Jill#!#Johnson#!#98.0
Jack#!#Green#!#67.0
Jennifer#!#Ross#!#75.7
Jacob#!#Brown#!#95.2
~~END~~


select top 100.00 percent lname from students
go
~~START~~
varchar
Doe
Smith
Johnson
Green
Ross
Brown
~~END~~


-- cleanup
drop table students
go
